Runtime/Controller/JointControl.cs

```diff
@@ -39,9 +39,10 @@ void Start()
             {
                 ArticulationDrive currentDrive = joint.xDrive;
                 float newTargetDelta = (int)direction * Time.fixedDeltaTime * speed;
-                if (newTargetDelta + currentDrive.target <= currentDrive.upperLimit && newTargetDelta + currentDrive.target >= currentDrive.lowerLimit){
+                if (((joint.linearLockX == ArticulationDofLock.LimitedMotion) && (newTargetDelta + currentDrive.target <= currentDrive.upperLimit && newTargetDelta + currentDrive.target >= currentDrive.lowerLimit)) || 
+                (joint.linearLockX == ArticulationDofLock.FreeMotion)){
                     currentDrive.target += newTargetDelta;
```

Comment:
Have one more question. If the `currentDrive.target`= 9, the `newTargetDelta` = 2, and the `currentdrive.upperLimit` = 10, should the target be set to 10 or do we still want to keep it 9?